core/src/main/resources/reference.conf

```diff
@@ -25,15 +25,13 @@ kanaloa {
     circuitBreaker {
       enabled = on
 
-      closeDuration = 1s
+      # Open duration per timeout, e.g. if the value is 10s and
+      # there are 3 consecutive timeouts, the circuit breaker will be open for 30 seconds
+      # if afterwards it gets another timeout, it will be open for another 40 seconds
+      openDurationBase = 10s
 
-      # The length of result logs, if the historyLength is 100 and the errorRateThreshold is 1
-      # The circuitBreaker will open after seeing 100 consecutive errors.
-      historyLength = 100
-
-      # CircuitBreaker opens when error rate within the history goes beyond this threshold
-      # value must be between 0 and 1
-      errorRateThreshold = 1.0
+      # CircuitBreaker opens for a worker when it sees consecutive timeouts at or above this threshold
+      timeoutCountThreshold = 3
```

Comment:
So, I just had a pretty wild idea.  This handling of timeouts seems very specific, ie: its either you handle timeouts in this manner, by counting timeouts, and then applying a multiplier, or nothing.  

One size may not fit all.

One idea is to provide a Backoff trait, which gets called for every timeout.  An instance would need to be created per worker, and it is guaranteed to be threadsafe code, since it would be called from within the Actor's receive function.  The idea would be that the user can provide their own implementation of how they want to handle timeouts.  IE: exponential backoff, etc.

The trait's interface would just be 
```scala
def handleTimeout(msg?) : Option[FiniteDuration]
```

We could easily provide a few different, basic implementations, including this algorithm here.